src/components/Timeperiod.jsx

```diff
@@ -134,4 +138,5 @@ export default connect(({ timeperiod: s, state }) => ({
     setOpen         : s.menu.setOpen,
     TimePeriodMenu  : s.TimePeriodMenu,
     timeUnit_display: s.timeUnit_display,
+    onGranularityChange: s.onChange,
```

Comment:
@MahboobehMohammadi it'd be better to choose another name to covey that it's adding the passed `onChange` function from parent to the `TimeperiodStore`.